jerry-core/jrt/jrt.h

```diff
@@ -109,6 +125,9 @@ void __noreturn jerry_unreachable (const char *file, const char *function, const
     jerry_unreachable (__FILE__, __func__, __LINE__); \
   } while (0)
 #else /* JERRY_NDEBUG */
+#if defined(_MSC_VER)
+#define JERRY_ASSERT(x) do {} while(0)
```

Comment:
Why is this variant necessary? The non-MSVC variant seems to be identical to this.

Reply:
I'm not sure I understand. There is no function in use here. `JERRY_UNUSED (x)` is just a macro defined some lines above and gets expanded by the preprocessor to `((void) (x))`. What does your linker report?

Reply:
do you see, ((void) (x));
If the x is a function call, the x got to be expanded and this function would be called.

Reply:
Actually, it would never be called because it is in an `if (false)` block. Until now, all compilers were able to optimize that away before it reached the linker. Do you have optimizations enabled?

If that does not help, It might still be easier/simpler to define `JERRY_UNUSED` to nothing in the case of MSVC.

Reply:
The more I think about it the less correct the MSVC compiler/linker behaviour seems. Even if the compiler is not optimizing the `if (false)` away, what can be there in `x` that's not linkable? We do test the debug configuration with assertions enabled and everything is fine both on Linux and on macOS. Which means that `x` will be evaluated, and if it contains function calls, they will be linked and called. This suggests me that we are not using anithing that's not there. So, the question still stands: what does your linker report / complain about?

Reply:
I am consider that if defines to empty， then its not a concrete statement along with a comma； anyway， empty is also acceptable

Reply:
Please, _do_ copy here what the linker reports as error.